### PR TITLE
python37Packages.cryptography: 2.5 -> 2.6.1

### PR DESCRIPTION
**Pull request by primeos**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by dtzWill — APPROVED** [review]
LGTM!

Looking at changelog for 2.6.1, this also helps use OpenSSL 1.1.x goodness once we switch over!

**Comment by worldofpeace** [comment]
@FRidh @dotlambda 

Is this good for merge into staging?

**Review by worldofpeace — APPROVED** [review]

**Comment by FRidh** [comment]
I have not yet run a job on Hydra as it's already overloaded with the release branch and such. I intend to run a job a week or so after the release, although actually I think everything will be fine.

**Comment by worldofpeace** [comment]
> I have not yet run a job on Hydra as it's already overloaded with the release branch and such. I intend to run a job a week or so after the release, although actually I think everything will be fine.

Ah I see, you feel it should be fine. But if there's any problems a hydra job would reveal it would probably be a good idea :+1: 

Thanks.

File: pkgs/development/python-modules/cryptography/default.nix [hunk 1]
```diff
@@ -20,18 +20,17 @@
...
   src = fetchPypi {
     inherit pname version;
-    sha256 = "00c4d7gvsymlaw0r13zrm32dcnarmpayjyrh65yymlmr6mrbcij9";
+    sha256 = "19iwz5avym5zl6jrrrkym1rdaa9h61j20ph4cswsqgv8xg5j3j16";
   };
 
   outputs = [ "out" "dev" ];
 
-  buildInputs = [ openssl cryptography_vectors ]
+  buildInputs = [ openssl ]
```
Comment (eadwu) [inline pkgs/development/python-modules/cryptography/default.nix]: Should `openssl` be pointed towards `openssl_1_1`? At least for `sourcehut` to work support for `ed25519` is needed which isn't provided by the default `openssl`.

```
    File "/nix/store/jszvzjmq5nx90dlw8d80hcyfwzzqzj7p-python3.7-cryptography-2.6.1/lib/python3.7/site-packages/cryptography/hazmat/primitives/asymmetric/ed25519.py", line 63, in from_private_bytes
      _Reasons.UNSUPPORTED_PUBLIC_KEY_ALGORITHM
  cryptography.exceptions.UnsupportedAlgorithm: ed25519 is not supported by this version of OpenSSL.
```

**Comment by dtzWill** [comment]
On Sat, 20 Apr 2019 14:43:16 -0700, Edmund Wu <notifications@github.com> wrote:
> 
> Should `openssl` be pointed towards `openssl_1_1`? At least for `sourcehut` to work support for `ed25519` is needed which isn't provided by the default `openssl`.
> 

Sounds good to me-- while we're at it,
when can we make this the default?

**Comment by primeos** [comment]
@eadwu @dtzWill I would prefer to change the OpenSSL version in another PR, optimally in #22357, to avoid the risk of breaking too many things at once and test it more extensively.

This PR is open for way too long now and should be pretty safe to merge (I'm not aware of any problems due to past updates of `cryptography` - apart from a few warning messages).
I'll run a few rebuilds and merge this if there aren't any problems (though I am not sure how useful that actually is as most failures would probably occur at runtime - apart from failures due to pinned versions, but that wasn't a problem in the past and shouldn't have changed).

**Comment by dtzWill** [comment]
On Mon, 22 Apr 2019 03:23:42 -0700, Michael Weiss <notifications@github.com> wrote:
> @eadwu @dtzWill I would prefer to change the OpenSSL version in another PR, optimally in #22357, to avoid the risk of breaking too many things at once and test it more extensively.


Oh, absolutely! I was just ... chatting about how we might want to do it
in general :).

Agreed that's a fight for a different PR.

**Comment by primeos** [comment]
Ok, great - yeah it's really time to switch to OpenSSL 1.1 in general...

I ran most rebuilds and didn't notice any failures that are related to this PR and this update should be fine :)

**Pull request merged**